signer/src/wsts_state_machine.rs

```diff
@@ -148,13 +148,19 @@ impl SignerStateMachine {
             wsts::util::encrypt(&self.0.network_private_key.to_bytes(), &encoded, rng)
                 .map_err(|_| error::Error::Encryption)?;
 
+        let signature_share_threshold: u16 = self
+            .threshold
+            .try_into()
+            .map_err(|_| Error::TypeConversion)?;
```

Comment:
Not a nit, just a question: 

I see you mostly do the type casts explicitly at the moment that something could fail instead of having a `From` definition for the error that the function returns. Why?

Reply:
I tend to do this for a few reasons:
1. The error returned from the `try_into` is not particularly informative, so having the dedicated variant is not super helpful.
2. I think implementing `From` for an error variant is not the right pattern for us to have. Instead the goal is to construct error variants with enough context data for us to debug when we encounter the error. And `From` impls, particularly `From` impls for foreign types, encourages the wrong behavior.
3. This particular error is not actually "real". The threshold is a `u32` type, but it is created from a `u16`, so the conversion back should always happen. So I don't think it's necessary to add context and all of that, I'd rather get rid of the error.

Because of (3) I could do an `as u16` cast, but I'm hesitant because it's kind of bad practice (as in we should be cautious of using them). Perhaps it's warranted in this case with a comment explaining why it is fine.